galaxy_ng/app/signals/handlers.py

```diff
@@ -56,12 +56,8 @@ def set_role_definitions(sender, **kwargs):
     if apps is None:
         from django.apps import apps
     role_prefix = f"{sender.label}."
-    desired_roles = {}
-    for viewset in GALAXY_VIEWSETS:
-        locked_roles = GALAXY_VIEWSETS[viewset]['LOCKED_ROLES']
-        if locked_roles is not None:
-            desired_roles.update(locked_roles or {})
-    adjust_roles(apps, role_prefix, desired_roles, verbosity=kwargs.get("verbosity", 1))
+    if GALAXY_LOCKED_ROLES is not None:
+        adjust_roles(apps, role_prefix, GALAXY_LOCKED_ROLES, verbosity=kwargs.get("verbosity", 1))
```

Comment:
Hmm, this might be a problem. `adjust_roles` is already called for every plugin by pulpcore. And it will enforce whatever it finds on the plugins viewsets (including deleting all roles if nothing was found).
I think, you can either define the locked roles on viewsets and just completely remove this hook, or use a different prefix.
A third option may be to adjust the plugins `AppConfig.ready`.